### PR TITLE
Fix missing functions and constants, part 2

### PR DESCRIPTION
**Pull request by darrenoh**

### CODE REVIEW

File: src/functions/user.php [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
 declare(strict_types=1);
 
+function user_access(string $string, ?AccountInterface $account = null): bool
+{
+    global $user;
+    if (!isset($account)) {
+        $account = $user;
+    }
+    return $account->hasPermission($string);
+}
```
Comment (mglaman) [inline src/functions/user.php:12]: This can be a candidate for its own PR to merge quickly.

File: src/functions/lock.php
Comment (mglaman) [inline src/functions/lock.php:1]: This can be a candidate for its own PR to merge quickly.

File: src/functions/db.php [hunk 2]
```diff
@@ -87,3 +87,8 @@ function db_condition(string $conjunction)
 {
     return DB::get()->condition($conjunction);
 }
+
+function db_table_exists(string $table): bool
+{
+    return DB::get()->schema()->tableExists($table);
+}
```
Comment (mglaman) [inline src/functions/db.php:94]: This can be a candidate for its own PR to merge quickly.

File: src/functions/cache.php
Comment (mglaman) [inline src/functions/cache.php:1]: This can be a candidate for its own PR to merge quickly.

**Review by mglaman — COMMENTED** [review]
I've flagged a few items that would be great to have in their own PR for quick merges

File: src/Render/RetrofitHtmlResponseAttachmentsProcessor.php [hunk 3]
```diff
@@ -31,7 +31,7 @@ public function processAttachments(AttachmentsInterface $response)
         }
         if (isset($attachments['js']) && is_array($attachments['js'])) {
             foreach ($attachments['js'] as $key => $item) {
-                if (isset($item['type'], $item['data']) && $item['type'] === 'setting') {
+                if (is_array($item) && isset($item['type'], $item['data']) && $item['type'] === 'setting') {
```
Comment (mglaman) [inline src/Render/RetrofitHtmlResponseAttachmentsProcessor.php:34]: What could the value be if it's not an array?

Comment (darrenoh) [reply src/Render/RetrofitHtmlResponseAttachmentsProcessor.php:34]: That's just to fix a phpstan message.

**Review by mglaman — COMMENTED** [review]
After #90 all that is left is `drupal_write_record`. Then the Role stuff should move to its own PR to sort out.

**Comment by mglaman** [comment]
@darrenoh with https://github.com/retrofit-drupal/retrofit/pull/92, can we close this PR and re-open it specifically for handling Roles?

**Comment by darrenoh** [comment]
Will open an new PR for user roles.
